src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IObjectCreationExpression.cs

```diff
@@ -9970,6 +9970,173 @@ class C1
+
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: MemberInitializerTest, IsInvalid, IsImplicit) (Syntax: 'i = new Mem ...  = 0, y++ }')
+              Left: 
+                ILocalReferenceOperation: i (IsDeclaration: True) (OperationKind.LocalReference, Type: MemberInitializerTest, IsInvalid, IsImplicit) (Syntax: 'i = new Mem ...  = 0, y++ }')
+              Right: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: MemberInitializerTest, IsInvalid, IsImplicit) (Syntax: 'new MemberI ...  = 0, y++ }')
+
+        Next (Regular) Block[B2]
+            Leaving: {R1}
+}
+
+Block[B2] - Exit
+    Predecessors: [B1]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<MethodDeclarationSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ObjectCreationFlow_69()
+        {
+            string source = @"
+#pragma warning disable 0169
+class A
+{
+    dynamic this[int x, int y]
+    {
+        get
+        {
+            return new A();
+        }
+    }
+
+    dynamic this[string x, string y]
+    {
+        get
+        {
+            throw null;
+        }
+    }
+
+    int X, Y, Z;
+
+    /*<bind>*/static void Main(A a, dynamic x)
+    {
+        a = new A {[y: x, x: x] = { X = 1, Y = 1, Z = 1 } };
+    }/*</bind>*/
+}
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            string expectedFlowGraph = @"
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (8)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'a')
+          Value: 
+            IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: A) (Syntax: 'a')
+
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'new A {[y:  ... , Z = 1 } }')
+          Value: 
+            IObjectCreationOperation (Constructor: A..ctor()) (OperationKind.ObjectCreation, Type: A) (Syntax: 'new A {[y:  ... , Z = 1 } }')
+              Arguments(0)
+              Initializer: 
+                null
+
+        IFlowCaptureOperation: 2 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'x')
+          Value: 
+            IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: dynamic) (Syntax: 'x')
+
+        IFlowCaptureOperation: 3 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'x')
+          Value: 
+            IParameterReferenceOperation: x (OperationKind.ParameterReference, Type: dynamic) (Syntax: 'x')
+
+        ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: dynamic) (Syntax: 'X = 1')
+          Left: 
+            IOperation:  (OperationKind.None, Type: null) (Syntax: 'X')
```

Comment:
>IOperation:  (OperationKind.None, Type: null) (Syntax: 'X') [](start = 12, length = 59)

The assignment target looks unexpected.

Reply:
[https://github.com/dotnet/roslyn/issues/27499](https://github.com/dotnet/roslyn/issues/27499). This is also the cause of the missing flow captures above. I can add a Children implementation for the node if you would like.

---
In reply to: [193594400](https://github.com/dotnet/roslyn/pull/27519#discussion_r193594400) [](ancestors = 193594400)

Reply:
@AlekseyTs, after some investigation, it looks like even in the simplified case we came up with offline, we're still encountering the BoundDynamicObjectInitializerMember in the inner initializer. The codepaths I added _are_ getting hit, which is why we have the unused flow captures above. It looks like it would be relatively simple to add support for simply transforming that node to an IDynamicMemberReferenceOperation, meaning we won't need any new nodes or information. I'm of the opinion I should take that approach, what are thoughts?

---
In reply to: [193594548](https://github.com/dotnet/roslyn/pull/27519#discussion_r193594548) [](ancestors = 193594548,193594400)

Reply:
FYI, I ended up doing this locally, the change required is very minimal. I'll push after I do a full test hook run to make sure we're still green there.

---
In reply to: [193598936](https://github.com/dotnet/roslyn/pull/27519#discussion_r193598936) [](ancestors = 193598936,193594548,193594400)